### PR TITLE
Do Not Use Empty Machines for Deployment if Locked for Series Upgrade

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This is the second of 2 patches for preventing deployment to machines that are in the upgrade-series workflow. It depends on https://github.com/juju/juju/pull/9387.

When deploying an application, or adding a unit, this prevents empty machines being recruited if they are locked for series upgrade.

## QA steps

- Bootstrap and add a trusty machine.
- `juju upgrade-series prepare 0 xenial`.
- `juju deploy redis` and check that a new machine is added.
- `juju add-unit redis` and check that a new machine is added.
- Machine 0 remains empty.

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1798097

### CODE REVIEW

File: state/unit.go [hunk 1]
```diff
@@ -2382,8 +2382,8 @@ var hasNoContainersTerm = bson.DocElem{
 		{{"children", bson.D{{"$exists", false}}}},
 	}}
 
-// findCleanMachineQuery returns a Mongo query to find clean (and possibly empty) machines with
-// characteristics matching the specified constraints.
+// findCleanMachineQuery returns a Mongo query to find clean (and maybe empty)
```
[the remark this answers is not in the document]
Comment (manadart) [reply state/unit.go]: No special reason. Just a synonym to get a line break at a nice spot inside 80 chars.

[the remark this answers is not in the document]
Comment (ExternalReality) [reply state/unit.go]: Ah, Ack.

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
Test timeout. $$merge$$

**Pull request merged**